packages/components/src/components/PipelineRun/PipelineRun.js

```diff
@@ -83,11 +117,26 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
           namespace: taskRunNamespace,
           annotations
         } = taskRun.metadata;
+
         const { reason, status: succeeded } = getStatus(taskRun);
+
         const { pipelineTaskName } = taskRunDetails[taskRunName] || {};
         const { params, resources: inputResources } = taskRun.spec.inputs;
         const { resources: outputResources } = taskRun.spec.outputs;
-        const steps = stepsStatus(task.spec.steps, taskRun.status.steps);
+
+        let steps = '';
+
+        if (!task.spec || !taskRun.status) {
```

Comment:
do you have a scenario that reliably gets us here? I'd like to see if there are other options as introducing fake data here can make this less flexible, especially when looking at how others might integrate the component

Reply:
Hmmmmmm, well I can reliably get taskRuns without a status coming in yeah, it's with five-ish rebuild runs in a row (you just keep rebuilding then clicking through to view that new one, then rebuild that, etc)

Same goes for the task steps, it's like the data just doesn't come in on time or we don't populate the store on time (and yet we do an async and wait on the response, so I'm guessing it's the websocket)...